compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -767,26 +767,38 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             });
         }
 
```

Comment:
This closure is the only "wacky" thing I introduced I'd say.

Reply:
yeah i would say too that this is the only slightly dissatisfying part of the pr. i wonder if we should separate pushing the trait bound and pushing the const predicate... 🤷

Reply:
I'll check if that'll make things clearer :+1: